daemon/exec.go

```diff
@@ -121,6 +123,13 @@ func (d *Daemon) ContainerExecCreate(name string, config *types.ExecConfig) (str
 	execConfig.Tty = config.Tty
 	execConfig.Privileged = config.Privileged
 	execConfig.User = config.User
+	execConfig.Env = []string{
+		"PATH=" + system.DefaultPathEnv,
+	}
+	if config.Tty {
+		execConfig.Env = append(execConfig.Env, "TERM=xterm")
```

Comment:
you should try to use global env TERM variable (maybe with security check or valid list) and fallback to TERM=xterm if missing